sockeye/data_io.py

```diff
@@ -374,9 +388,12 @@ def __init__(self,
                  target_sentences: List[List[int]],
                  buckets: List[Tuple[int, int]],
                  batch_size: int,
+                 batch_by_words: bool,
+                 batch_num_devices: int,
                  eos_id: int,
                  pad_id: int,
                  unk_id: int,
+                 bucket_batch_sizes: Optional[List[List[int]]] = None,
```

Comment:
these are always lists of tuples with 2 elements, right? You could either use: `Optional[List[Tuple[int, int]]]` or potentially even put `(batch_size_seq, batch_size_word)` into a namedtuple.